docroot/modules/custom/va_gov_vamc/src/ServiceLocationMigration.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+namespace Drupal\va_gov_vamc;
+
+use Drupal\paragraphs\Entity\Paragraph;
+use Psr\Log\LogLevel;
+
+/**
+ * Single use class to migrate from facility service node to service location.
+ *
+ * This is not a service, it is only needed one time so DI can not be used.
+ */
+class ServiceLocationMigration {
+
+
+  /**
+   * The current facility service node being processed.
+   *
+   * @var \Drupal\node\NodeInterface
+   */
+  protected $facilityService;
+
+  /**
+   * The current service_location paragraph being processed.
+   *
+   * @var \Drupal\paragraphs\ParagraphInterface
+   */
+  protected $serviceLocation;
+
+  /**
+   * Constructor for ServiceLocationMigration.
+   *
+   * @param array $sandbox
+   *   Sandbox variable for keeping state during batches.
+   */
+  public function __construct(array &$sandbox) {
+    require_once __DIR__ . '/../../../../../scripts/content/script-library.php';
+    $processed_nids = '';
+    $batch_size = 25;
+    $node_storage = get_node_storage();
+    $nids = array_slice($sandbox['items_to_process'], 0, $batch_size, TRUE);
+    $facility_services = (empty($nids)) ? [] : $node_storage->loadMultiple(array_values($nids));
```

Comment:
@omahane and @Becapa Because of the new state keeping logic in https://github.com/department-of-veterans-affairs/va.gov-cms/pull/17236  this will likely fall apart a bit because the state is tracking per item, not per batch.
The state does not know how to reference which batch of 25 was just finished and would not really be able to pick up where it left off.   Also the output would likely be jibberish.

I suggest this be changed to remove the batch size and just grab the top item in the array.  Something like this.  (not tested)

```suggestion
    $node_storage = get_node_storage();
    $nid_to_load = reset($sandbox['items_to_process']);
    $facility_services = (empty($nid_to_load)) ? [] : $node_storage->loadMultiple(array_values($nid_to_load));
```